### ISSUE TITLE
minifier: removes "0"-prefixed octal literals even in strict mode

### ISSUE THREAD

**Issue by sapphi-red** [oxc-project/oxc#8156]
`0o0` should not be removed because it throws an error in strict mode.

[playground](https://playground.oxc.rs/#eNpFjz1vg0AMhv8K8tRKUUUrdaF75g4dWS5XQ086bGSbpAjx3+uDkE7+eO33sReI0EA3UbTEVHXMT8/V0lJVtTApVmqSorXwUVp17WFtCU7A0CwgE5WgM1n4hcZkwhPkRAZNF7J6oZFHPBSdhwvnozIJpB3LcB9eTzAGUZTNkieJ+DWXZTeRNJpD9+45ZaQwFMVQ7cXUpVHQV6/4GQRJ/y3LNZul5w/izrAgPfqpgPpWv76DT0T+xh7LU14MiVKX9tnbT3LWGKJT769FHgrUWfs/Q6A+H/LmRSacz5lvxeGKcmF96Ov6B+OxhBQ=)

```js
(function foo() {
  "use strict";
  00;
})() // SyntaxError: Octal literals are not allowed in strict mode.
```
```js
(function foo() {
  "use strict";
})() // No error
```

**Comment by Boshen** [comment]
Is this a real world usage? I'm reluctant to fix these kind of problems otherwise 😅

**Comment by sapphi-red** [comment]
No, it's not a real world usage. I noticed about this case and make this issue just in case.
